earth_enterprise/src/common/khCache.h

```diff
@@ -322,6 +326,29 @@ class khCache {
       }
     }
     CheckListInvariant();
+    TimePoint finishTime = std::chrono::high_resolution_clock::now();
+    // Log when the cache size is exceeded
+    if (TooMuchMemory() || TooManyObjects()) {
```

Comment:
Looks like this is being called after the pruning is already done. Seems like `TooMuchMemory()` and/or `TooManyObjects()` would return false after a successful pruning, preventing `LogPruneTime()` from being called.

Reply:
If `use_count() > 1` is true for lots of items, the Prune function can finish with the cache oversize. That's when we want to print the log message.

Reply:
Okay, I misunderstood the intent. I thought it was to show a log any time the cache size was exceeded and about to be pruned. But the goal is to log when the cache limit is still exceeded after being pruned. Got it.